Framework/Algorithms/test/PolarizationCorrectionFredrikzeTest.h

```diff
@@ -362,4 +362,125 @@ class PolarizationCorrectionFredrikzeTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(out3->y(0)[0], 0.7, 1e-14);
     TS_ASSERT_DELTA(out4->y(0)[0], 0.6, 1e-14);
   }
+
+  void test_valid_spin_state_order_for_PA() {
+    auto groupWS = std::make_shared<WorkspaceGroup>();
+    groupWS->addWorkspace(create1DWorkspace(4, 1, 1));
+    groupWS->addWorkspace(create1DWorkspace(4, 1, 1));
+    groupWS->addWorkspace(create1DWorkspace(4, 1, 1));
+    groupWS->addWorkspace(create1DWorkspace(4, 1, 1));
+
+    PolarizationCorrectionFredrikze alg;
+    alg.setChild(true);
+    alg.setRethrows(true);
+    alg.initialize();
+    alg.setProperty("InputWorkspace", groupWS);
+    alg.setPropertyValue("OutputWorkspace", "dummy");
+    alg.setProperty("PolarizationAnalysis", "PA");
```

Comment:
This group workspace creation and algorithm initialisation could probably be consolidated into their own methods, they're repeated a few times in the new tests. You could have an if statement in there for producing the smaller PNR ones as well.